edit-post/components/options-modal/index.js

```diff
@@ -0,0 +1,179 @@
+	setIsPublishSidebarEnabled,
+	areTipsEnabled,
+	setAreTipsEnabled,
+	isStatusPanelEnabled,
+	setIsStatusPanelEnabled,
+	isCategoriesPanelEnabled,
+	setIsCategoriesPanelEnabled,
+	isTagsPanelEnabled,
+	setIsTagsPanelEnabled,
+	isFeaturedImagePanelEnabled,
+	setIsFeaturedImagePanelEnabled,
+	isExcerptPanelEnabled,
+	setIsExcerptPanelEnabled,
+	isDiscussionPanelEnabled,
+	setIsDiscussionPanelEnabled,
+} ) {
+	if ( ! isModalActive ) {
+		return null;
+	}
+
+	const title = (
+		<span className="edit-post-options__title">
+			{ __( 'Options' ) }
+		</span>
+	);
+
+	return (
+		<Modal
+			className="edit-post-options"
+			title={ title }
+			closeLabel={ __( 'Close' ) }
+			onRequestClose={ toggleModal }
+		>
+			<Section title={ __( 'General' ) }>
+				<CheckboxControl
+					className="edit-post-options__option"
+					label={ __( 'Enable Pre-publish Checks' ) }
+					checked={ isPublishSidebarEnabled }
+					onChange={ setIsPublishSidebarEnabled }
+				/>
+				<CheckboxControl
+					className="edit-post-options__option"
+					label={ __( 'Enable Tips' ) }
+					checked={ areTipsEnabled }
+					onChange={ setAreTipsEnabled }
+				/>
+			</Section>
+			<Section title={ __( 'Document Panels' ) }>
+				<CheckboxControl
+					className="edit-post-options__option"
+					label={ __( 'Status & Visibility' ) }
+					checked={ isStatusPanelEnabled }
+					onChange={ setIsStatusPanelEnabled }
+				/>
+				<CheckboxControl
+					className="edit-post-options__option"
+					label={ __( 'Categories' ) }
+					checked={ isCategoriesPanelEnabled }
+					onChange={ setIsCategoriesPanelEnabled }
+				/>
+				<CheckboxControl
+					className="edit-post-options__option"
+					label={ __( 'Tags' ) }
+					checked={ isTagsPanelEnabled }
+					onChange={ setIsTagsPanelEnabled }
+				/>
+				<CheckboxControl
+					className="edit-post-options__option"
+					label={ __( 'Featured Image' ) }
+					checked={ isFeaturedImagePanelEnabled }
+					onChange={ setIsFeaturedImagePanelEnabled }
+				/>
+				<CheckboxControl
+					className="edit-post-options__option"
+					label={ __( 'Excerpt' ) }
+					checked={ isExcerptPanelEnabled }
+					onChange={ setIsExcerptPanelEnabled }
+				/>
+				<CheckboxControl
+					className="edit-post-options__option"
+					label={ __( 'Discussion' ) }
+					checked={ isDiscussionPanelEnabled }
+					onChange={ setIsDiscussionPanelEnabled }
+				/>
+			</Section>
+		</Modal>
+	);
+}
+
+export default compose( [
+	withSelect( ( select ) => {
+		const { isModalActive, isEditorSidebarPanelEnabled } = select( 'core/edit-post' );
+		const { isPublishSidebarEnabled } = select( 'core/editor' );
+		const { areTipsEnabled } = select( 'core/nux' );
+
+		return {
+			isModalActive: isModalActive( MODAL_NAME ),
+			isPublishSidebarEnabled: isPublishSidebarEnabled(),
+			areTipsEnabled: areTipsEnabled(),
+			isStatusPanelEnabled: isEditorSidebarPanelEnabled( 'post-status' ),
+			isCategoriesPanelEnabled: isEditorSidebarPanelEnabled( 'taxonomy-panel-category' ),
```

Comment:
Users can define their own taxonomies so we'll need to add some weird dynamics here.